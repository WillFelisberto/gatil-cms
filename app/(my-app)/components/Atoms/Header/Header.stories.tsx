import type { Meta, StoryObj } from "@storybook/react";
import { Header } from "./Header";

const meta: Meta<typeof Header> = {
  title: "Atoms/Header", // Organização no painel do Storybook
  component: Header,
  parameters: {
    layout: "fullscreen", // Mostra em tela cheia
  },
  tags: ["autodocs"], // Permite geração automática de documentação
};

export default meta;
type Story = StoryObj<typeof Header>;

export const Default: Story = {
  args: {},
};
