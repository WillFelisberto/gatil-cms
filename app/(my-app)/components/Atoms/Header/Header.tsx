'use client';

import { Menu } from 'lucide-react';
import { usePathname } from 'next/navigation';
import { useState } from 'react';

type MenuItem = {
  label: string;
  href: string;
};

type HeaderProps = {
  menuItems: MenuItem[];
};

export const Header = ({ menuItems }: HeaderProps) => {
  const pathname = usePathname();
  const [isOpen, setIsOpen] = useState(false);

  return (
    <header className="bg-[#013274] text-white" role="banner" data-testid="header">
      <div className="max-w-7xl mx-auto px-4 flex justify-between items-center h-16 md:h-[91px]">
        <a href="/" aria-label="Ir para a pÃ¡gina inicial" className="flex items-center space-x-2">
          <img
            src="/logo-gatil.webp"
            alt="Gatil dos Resgatados - Logo"
            className="w-[230px] h-auto"
            loading="eager"
            fetchPriority="high"
            data-testid="logo"
          />
        </a>

        <nav
          className="hidden md:flex w-full justify-around text-sm font-semibold"
          role="navigation"
          aria-label="Menu principal"
        >
          {menuItems.map((link) => {
            const isActive = pathname === link.href;
            return (
              <a
                key={link.label}
                href={link.href}
                aria-current={isActive ? 'page' : undefined}
                aria-label={`Ir para ${link.label}`}
                className="relative text-center pt-2 w-full text-base group"
                data-testid={`nav-link-${link.label.toLowerCase()}`}
              >
                <span
                  className={`transition-colors duration-300 ${
                    isActive ? 'text-blue-300' : 'text-white group-hover:text-blue-300'
                  }`}
                >
                  {link.label}
                </span>
                <span
                  className={`absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 ${
                    isActive ? 'scale-x-100' : ''
                  }`}
                  style={{ width: '100%' }}
                ></span>
              </a>
            );
          })}
        </nav>

        <button
          onClick={() => setIsOpen(!isOpen)}
          className="md:hidden"
          aria-label={isOpen ? 'Fechar menu' : 'Abrir menu'}
          aria-expanded={isOpen}
          aria-controls="mobile-menu"
          data-testid="menu-button"
        >
          <Menu className="h-6 w-6 text-white" />
        </button>
      </div>

      {isOpen && (
        <div
          id="mobile-menu"
          className="md:hidden px-4 pt-0 pb-4 space-y-2 bg-[#002f6c] animate-fade-slide-down"
          role="menu"
          aria-label="Menu mobile"
        >
          {menuItems.map((link) => {
            const isActive = pathname === link.href;
            return (
              <a
                key={link.label}
                href={link.href}
                role="menuitem"
                aria-current={isActive ? 'page' : undefined}
                aria-label={`Ir para ${link.label}`}
                className={`block font-medium border-l-4 pl-2 ${
                  isActive
                    ? 'text-blue-300 border-blue-300'
                    : 'text-white border-transparent hover:text-blue-300 hover:border-blue-300'
                }`}
                data-testid={`mobile-link-${link.label.toLowerCase()}`}
              >
                {link.label}
              </a>
            );
          })}
        </div>
      )}
    </header>
  );
};
