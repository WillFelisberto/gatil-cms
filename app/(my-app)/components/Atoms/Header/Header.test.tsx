import { fireEvent, render, screen } from '@testing-library/react';

import { Header } from './Header';

describe('<Header />', () => {
  beforeEach(() => {
    render(<Header />);
  });

  it('Should render the logo with correct alt text', () => {
    const logo = screen.getByTestId('logo');
    const header = screen.getByTestId('header');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('alt', 'Gatil dos Resgatados - Logo');

    expect(header).toMatchSnapshot();
  });

  it('Should render all desktop navigation links', () => {
    const labels = ['Início', 'Apadrinhe', 'Colabore', 'Contato', 'Projeto', 'Adote'];
    labels.forEach((label) => {
      const link = screen.getByTestId(`nav-link-${label.toLowerCase()}`);
      expect(link).toBeInTheDocument();
      expect(link).toHaveTextContent(label);
    });
  });

  it('Should render the mobile menu button', () => {
    const menuButton = screen.getByTestId('menu-button');
    expect(menuButton).toBeInTheDocument();
    expect(menuButton).toHaveAttribute('aria-label', 'Abrir menu');
    expect(menuButton).toHaveAttribute('aria-expanded', 'false');
  });

  it('Should open the mobile menu when the button is clicked', () => {
    const menuButton = screen.getByTestId('menu-button');
    fireEvent.click(menuButton);

    const mobileMenu = screen.getByRole('menu');
    expect(mobileMenu).toBeInTheDocument();

    const mobileLink = screen.getByTestId('mobile-link-início');
    expect(mobileLink).toBeInTheDocument();

    expect(menuButton).toHaveAttribute('aria-expanded', 'true');
    expect(menuButton).toHaveAttribute('aria-label', 'Fechar menu');
  });

  it('Should close the mobile menu when the button is clicked again', () => {
    const menuButton = screen.getByTestId('menu-button');

    fireEvent.click(menuButton); // open the mobile menu
    expect(screen.getByRole('menu')).toBeInTheDocument(); // menu is visible

    fireEvent.click(menuButton); // close the mobile menu
    expect(screen.queryByRole('menu')).not.toBeInTheDocument(); // menu is gone

    expect(menuButton).toHaveAttribute('aria-expanded', 'false');
    expect(menuButton).toHaveAttribute('aria-label', 'Abrir menu');
  });

  it('Should highlight the active link with aria-current and correct class', () => {
    const activeLink = screen.getByTestId('nav-link-início');
    expect(activeLink).toHaveAttribute('aria-current', 'page');

    const span = activeLink.querySelector('span');
    expect(span).toHaveClass('text-blue-300');
  });
});
