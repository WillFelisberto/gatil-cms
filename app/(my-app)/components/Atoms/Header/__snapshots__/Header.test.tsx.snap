// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render the logo with correct alt text 1`] = `
<header
  class="bg-[#013274] text-white"
  data-testid="header"
  role="banner"
>
  <div
    class="max-w-7xl mx-auto px-4 flex justify-between items-center h-16 md:h-[91px]"
  >
    <a
      aria-label="Ir para a página inicial"
      class="flex items-center space-x-2"
      href="/"
    >
      <img
        alt="Gatil dos Resgatados - Logo"
        class="w-[230px] h-auto"
        data-testid="logo"
        fetchpriority="high"
        loading="eager"
        src="/logo-gatil.webp"
      />
    </a>
    <nav
      aria-label="Menu principal"
      class="hidden md:flex w-full justify-around text-sm font-semibold"
      role="navigation"
    >
      <a
        aria-current="page"
        aria-label="Ir para Início"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-início"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-blue-300"
        >
          Início
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 scale-x-100"
          style="width: 100%;"
        />
      </a>
      <a
        aria-label="Ir para Apadrinhe"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-apadrinhe"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-white group-hover:text-blue-300"
        >
          Apadrinhe
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 "
          style="width: 100%;"
        />
      </a>
      <a
        aria-label="Ir para Colabore"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-colabore"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-white group-hover:text-blue-300"
        >
          Colabore
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 "
          style="width: 100%;"
        />
      </a>
      <a
        aria-label="Ir para Contato"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-contato"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-white group-hover:text-blue-300"
        >
          Contato
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 "
          style="width: 100%;"
        />
      </a>
      <a
        aria-label="Ir para Projeto"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-projeto"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-white group-hover:text-blue-300"
        >
          Projeto
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 "
          style="width: 100%;"
        />
      </a>
      <a
        aria-label="Ir para Adote"
        class="relative text-center pt-2 w-full text-base group"
        data-testid="nav-link-adote"
        href="#"
      >
        <span
          class="transition-colors duration-300 text-white group-hover:text-blue-300"
        >
          Adote
        </span>
        <span
          class="absolute left-0 top-0 h-[2px] bg-blue-300 transition-all duration-300 origin-left scale-x-0 group-hover:scale-x-100 "
          style="width: 100%;"
        />
      </a>
    </nav>
    <button
      aria-controls="mobile-menu"
      aria-expanded="false"
      aria-label="Abrir menu"
      class="md:hidden"
      data-testid="menu-button"
    >
      <svg
        class="lucide lucide-menu h-6 w-6 text-white"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="20"
          y1="12"
          y2="12"
        />
        <line
          x1="4"
          x2="20"
          y1="6"
          y2="6"
        />
        <line
          x1="4"
          x2="20"
          y1="18"
          y2="18"
        />
      </svg>
    </button>
  </div>
</header>
`;
